sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -962,16 +1163,40 @@ pi_result piEnqueueMemBufferRead(pi_queue Queue, pi_mem Src,
                                  pi_uint32 NumEventsInWaitList,
                                  const pi_event *EventWaitList,
                                  pi_event *Event) {
-  (void)Queue;
-  (void)Src;
-  (void)BlockingRead;
-  (void)Offset;
-  (void)Size;
-  (void)Dst;
-  (void)NumEventsInWaitList;
-  (void)EventWaitList;
-  (void)Event;
-  DIE_NO_IMPLEMENTATION;
+  /// TODO : Support Blocked read, 'Queue' handling
+  if (BlockingRead) {
+    assert(false && "ESIMD_CPU support for blocking piEnqueueMemBufferRead is NYI");
+  }
+  if (NumEventsInWaitList != 0) {
+    return PI_INVALID_EVENT_WAIT_LIST;
+  }
+
+  _pi_buffer *buf = static_cast<_pi_buffer *>(Src);
+
+  int Status =
+      buf->CmBufferPtr->ReadSurface(reinterpret_cast<unsigned char *>(Dst),
+                                    nullptr, // event
+                                    static_cast<uint64_t>(Size));
+
+  if (Status != cm_support::CM_SUCCESS) {
+    return PI_INVALID_MEM_OBJECT;
+  }
+
+  if (Event) {
+    try {
+      *Event = new _pi_event();
+    } catch (const std::bad_alloc &) {
+      return PI_OUT_OF_HOST_MEMORY;
+    } catch (...) {
+      return PI_ERROR_UNKNOWN;
+    }
+
+    // At this point, CM already completed buffer-read (ReadSurface)
+    // operation. Therefore, 'event' corresponding to this operation
+    // is marked as dummy one and ignored during events-waiting.
+    (*Event)->IsDummyEvent = true;
```

Comment:
@romanovvlad , one question here. Can I return `nullptr` for `*Event` instead of creating and returning a 'dummy' event that is being ignored in `piEventsWait()`? As commented here, CM already completed `ReadSurface` operation using CPU (i.e. no off-loading to device like actual GPU/FPGA) therefore returning dummy event makes no point here.

Reply:
`nullptr` can be treated as a special value, so returning it as an event for successful operation is not recommended.
To avoid redundant allocations would could return some predefined but not `nullptr` value, for example `0xabcd`.
I'm not sure it's right direction since there are event queries that fully "conformant" plugin should implement, for example - `piEventGetProfilingInfo`.

Reply:
I see. Then I'll keep this 'IsDummyEvent` attribute. (Or I can rename it showing that the event is already done by CPU.) Thanks,